pandas/tests/test_index.py

```diff
@@ -2447,6 +2443,567 @@ def test_slice_keep_name(self):
         idx = Int64Index([1, 2], name='asdf')
         self.assertEqual(idx.name, idx[1:].name)
 
+class TestRangeIndex(Numeric, tm.TestCase):
+    _holder = RangeIndex
+    ### what does the following do?
+    #_multiprocess_can_split_ = True
+
+    def setUp(self):
+        self.indices = dict(index = RangeIndex(0, 20, 2))
+        self.setup_indices()
+
+    def create_index(self):
+        return RangeIndex(5)
+
+    def test_too_many_names(self):
+        def testit():
+            self.index.names = ["roger", "harold"]
+        assertRaisesRegexp(ValueError, "^Length", testit)
+
+    def test_constructor(self):
+        index = RangeIndex(5)
+        expected = np.arange(5, dtype=np.int64)
+        self.assert_numpy_array_equal(index, expected)
```

Comment:
Point taken on class & attribute checking.

However `RangeIndex` is currently by definition `int64`. The constructor does/should not allow specification of a `dtype`:
`def __new__(cls, start=None, stop=None, step=None, name=None, fastpath=False)`
Or am I overlooking something?

Or are you saying that I should allow `dtype` specification? I am not sure I am really getting the point here. I could make `dtype` variable. But then I would need to check for all inputs not only that they are `int64` but that they are the right `int` type. Is that worth the effort?